cmake/mex.cmake

```diff
@@ -225,21 +225,15 @@ function(mex_setup)
 
 
   # todo: handle C separately from CXX?
-  set (MEX_COMPILE_FLAGS "${MEX_INCLUDE} ${MEX_CXXFLAGS} ${MEX_DEFINES} ${MEX_MATLABMEX} ${MEX_CXX_ARGUMENTS}")
```

Comment:
MEX_CXXFLAGS includes MEX_DEFINES on all platforms, which includes MEX_MATLABMEX. Another was added below, so at the very least there were four `-DMATLAB_MEX_FILE` flags on each target...